crates/ev-cli/src/commands/function/deploy.rs

```diff
@@ -0,0 +1,237 @@
+/// Deploy a function
+#[derive(Parser, Debug)]
+pub struct DeployArgs {
+    /// Don't track the Function deployment status
+    #[arg(short, long, default_value_t = false)]
+    pub background: bool,
+}
+
+#[derive(Error, Debug)]
+pub enum DeployError {
+    #[error("An error occurred while fetching the Functions for your app: {0}")]
+    FetchAppFunctions(ApiError),
+    #[error("No name field found in function.toml. Make sure the function.toml in the current directory contains a name field.")]
+    MissingNameField,
+    #[error(transparent)]
+    ValidationError(#[from] validators::ValidationError),
+    #[error(transparent)]
+    FsError(#[from] crate::fs::FsError),
+    #[error("{1} was deprecated on {0}. ")]
+    VersionDeprecated(NaiveDate, String),
+    #[error("{1} will be deprecated on {0}.")]
+    VersionWillBeDeprecated(NaiveDate, String),
+    #[error("An error occured creating your Function record: {0}")]
+    RecordCreate(ApiError),
+    #[error("The zipped Function source was not found.")]
+    ZipNotFound,
+    #[error("An error occurred uploading the zipped Function source: {0}")]
+    FunctionUpload(ApiError),
+    #[error("An error occured deploying your Function")]
+    DeploymentFailed,
+    #[error("An error occurred fetching the deployment status of your Function")]
+    DeploymentStatusFetch(ApiError),
+    #[error("An error occured deploying your Function. The deployment was found to be in a cancelled state.")]
+    DeploymentCancelled,
+}
+
+impl CmdOutput for DeployError {
+    fn code(&self) -> String {
+        match self {
+            DeployError::FetchAppFunctions(_) => "function-fetch-functions-error",
+            DeployError::MissingNameField => "function-missing-name-error",
+            DeployError::ValidationError(_) => "function-validation-error",
+            DeployError::FsError(_) => "function-fs-error",
+            DeployError::VersionDeprecated(_, _) => "function-version-deprecated-error",
+            DeployError::VersionWillBeDeprecated(_, _) => {
+                "function-version-will-be-deprecated-error"
+            }
+            DeployError::RecordCreate(_) => "function-record-create-error",
+            DeployError::ZipNotFound => "function-zip-not-found-error",
+            DeployError::FunctionUpload(_) => "function-upload-error",
+            DeployError::DeploymentFailed | DeployError::DeploymentCancelled => {
+                "function-deployment-error"
+            }
+            DeployError::DeploymentStatusFetch(_) => "function-deployment-status-fetch-error",
+        }
+        .to_string()
+    }
+
+    fn exitcode(&self) -> crate::errors::ExitCode {
+        crate::errors::GENERAL
+    }
+}
+
+#[derive(strum_macros::Display, Debug)]
+pub enum DeployMessage {
+    #[strum(to_string = "Function ({uuid}) Deployed Succesfully")]
+    Deployed { uuid: String },
+    #[strum(
+        to_string = "Function deployment initiated successfully. Deployment will continue in the background."
```

Comment:
Worth adding something to the effect of "You can watch the deployment's progress in the Evervault dashboard."